src/controllers/Realtime.ts

```diff
@@ -60,6 +60,10 @@ export class RealtimeController extends BaseController {
     this.kuzzle.on('disconnected', () => this.saveSubscriptions());
     this.kuzzle.on('networkError', () => this.saveSubscriptions());
     this.kuzzle.on('reconnected', () => this.resubscribe());
+    this.kuzzle.on('reAuthenticated', () => {
+      this.saveSubscriptions();
```

Comment:
Why do you save the subscriptions? I understand you want to keep them safe in case of error during reAuth but I think this could lead to subscriptions duplication if reAuth failed then success. Don't you think so?

Reply:
This will not lead to subscription duplication since this event is called after the authenticator is called, meaning that an `auth:login` operation has been done.
Since a `auth:login` has been done the subscriptions are not existing anymore in Kuzzle, we do the save the subscriptions of the SDK to then do a resubscribe directly to renew the subscriptions.